server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -327,6 +336,8 @@ public BarrageMessageProducer(
 
         this.propagationRowSet = RowSetFactory.empty();
         this.updateIntervalMs = updateIntervalMs;
+        this.operationInitializer = operationInitializer;
+        this.updatePropagationJob = new UpdatePropagationJob();
```

Comment:
Arguably, it would be better to pass the operation initializer into the UpdatePropagationJob's constructor and to not keep a reference to it on the BMP itself. (Since the prop job creates an execution context and never looks at the operation initializer again.)